presto-main/src/main/java/com/facebook/presto/operator/HashBuilderOperator.java

```diff
@@ -316,6 +321,7 @@ public State getState()
                 return spilledLookupSourceHandle.getDisposeRequested();
 
             case DISPOSED:
+            case CLOSED:
```

Comment:
after close we should probably return NOT_BLOCKED...
in fact, I have problem recalling why i return `lookupSourceFactoryDestroyed` in `DISPOSED` state.. Maybe because this describes better what is the actual control from between components, but it doesn't look necessary anyway.